datafusion/core/src/dataframe/parquet.rs

```diff
@@ -166,4 +166,47 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn write_parquet_with_small_rg_size() -> Result<()> {
+        let mut test_df = test_util::test_table().await?;
+        // make the test data larger so there are multiple batches
+        for _ in 0..7 {
+            test_df = test_df.clone().union(test_df)?;
```

Comment:
When I ran this test it takes more than 22 seconds on my laptop. I wonder if we really need to generate so much data -- maybe we can try slicing up the batch (or else maybe use larger rg_sizes)

```shell
$ cargo test --lib -p datafusion -- write_parquet_with_small_rg_size
...
    Finished test [unoptimized + debuginfo] target(s) in 0.16s
     Running unittests src/lib.rs (target/debug/deps/datafusion-4cbfc61ad6017be4)

running 1 test
test dataframe::parquet::tests::write_parquet_with_small_rg_size ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 651 filtered out; finished in 22.31s

```

Reply:
We should be able to trigger the issue with less data by lowering `execution.batch_size` to something small like 10 rows.

Reply:
The test with a batch size of 10 still panics on main and passes in this PR, but runs in 0.41 seconds.